### DISCUSSION TITLE
Is there version of AsyncESPWebserver that supports AsyncTCP_SSL?

### DISCUSSION THREAD

**Discussion by strocode** [khoih-prog/AsyncTCP_SSL#4] [Q&A]
Hi,

This repo looks like the glorious future of AsyncTCP. Is it? Does AsyncESPWebserver work with it yet? Are the any instructions on how to do that somewhere?

**Comment by khoih-prog** [comment]
Hi @strocode 

Thanks for your interest in the library, which has been used in [**AsyncHTTPSRequest_Generic**](https://github.com/khoih-prog/AsyncHTTPSRequest_Generic)

I'm not aware of any **SSL** [**ESPAsyncWebServer**](https://github.com/me-no-dev/ESPAsyncWebServer)-related library yet, but guess there are some out there. As I haven't spend any time on this subject, I can't say anything about instructions. Please research / try yourself by replacing `AsyncTCP` with `AsyncTCP_SSL`. 

https://github.com/me-no-dev/ESPAsyncWebServer/blob/master/src/ESPAsyncWebServer.h#L33

Check related good link [**Workarounds to make SSL/TLS work** #899](https://github.com/me-no-dev/ESPAsyncWebServer/issues/899)

Post the test results, fork / create a new library, etc. so that we all can benefit from your work.

The **big issue** I can imagine is that **ESP32 don't have enough power** to provide ESPAsyncWebServer, with **TLS/SSL**, which requires lot of calculations / memory / speed.

It's better to use more powerful system, such as **Raspberry Pi 3/4**, etc., to handle this AsyncSSLWebServer task.

**Comment by strocode** [comment]
OK thanks Khoi,

I think there are some ESP32 SSL servers out there, e.g. [1,2] but I really
like the ESPAsyncWebServer framework. I might try to hack at it to see how
well it works.

Thanks for the reply.

Keith

[1]
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/protocols/esp_https_server.html
[2] https://github.com/fhessel/esp32_https_server



On Thu, 21 Jul 2022 at 03:38, Khoi Hoang ***@***.***> wrote:

> Hi @strocode <https://github.com/strocode>
>
> Thanks for your interest in the library, which has been used in
> *AsyncHTTPSRequest_Generic*
> <https://github.com/khoih-prog/AsyncHTTPSRequest_Generic>
>
> I'm not aware of any *SSL* *ESPAsyncWebServer*
> <https://github.com/me-no-dev/ESPAsyncWebServer>-related library yet, but
> guess there are some out there. As I haven't spend any time on this
> subject, I can't say anything about instructions. Please research / try
> yourself by replacing AsyncTCP with AsyncTCP_SSL.
>
>
> https://github.com/me-no-dev/ESPAsyncWebServer/blob/master/src/ESPAsyncWebServer.h#L33
>
> Post the test results, fork / create a new library, etc. so that we all
> can benefit from your work.
>
> The *big issue* I can imagine is that *ESP32 don't have enough power* to
> provide ESPAsyncWebServer, with *TLS/SSL*, which requires lot of
> calculations / memory / speed.
>
> It's better to use more powerful system, such as *Raspberry Pi 3/4*,
> etc., to handle this AsyncSSLWebServer task.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/khoih-prog/AsyncTCP_SSL/discussions/4#discussioncomment-3191561>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOS22NEB2EDEDZAK3WQDUTVVA2PJANCNFSM532YSLNQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Keith Bannister
